conda-store-server/conda_store_server/__init__.py

```diff
@@ -6,5 +7,81 @@
 CONDA_STORE_DIR = Path.home() / ".conda-store"
 
 
-# Default build_key_version. Must be None here, initialized from the config file
-_BUILD_KEY_VERSION = None
+class BuildKey:
+    # Avoids a cyclic dependency between the orm module and the module defining
+    # CondaStore.build_key_version. Because the orm module is loaded early on
+    # startup, we want to delay initialization of the Build.build_key_version
+    # field until CondaStore.build_key_version has been read from the config.
+
+    # Default version, must be None here. Initialized in CondaStore.build_key_version
+    _current_version = None
+
+    _version2_hash_size = 8
+
+    def _version1_fmt(build: "Build"):  # noqa: F821
+        datetime_format = "%Y%m%d-%H%M%S-%f"
+        hash = build.specification.sha256
+        timestamp = build.scheduled_on.strftime(datetime_format)
+        id = build.id
+        name = build.specification.name
+        return f"{hash}-{timestamp}-{id}-{name}"
+
+    def _version2_fmt(build: "Build"):  # noqa: F821
+        tzinfo = datetime.timezone.utc
+        hash = build.specification.sha256[: BuildKey._version2_hash_size]
+        timestamp = int(build.scheduled_on.replace(tzinfo=tzinfo).timestamp())
+        id = build.id
+        name = build.specification.name
+        return f"{hash}-{timestamp}-{id}-{name}"
```

Comment:
Note: without `tzinfo`, this would vary between machines because `timestamp` uses the system clock by default, which might be not in UTC.